### PR TITLE
Fixed lint and build errors on Linux

### PR DESCRIPTION
**Pull request by TekuConcept**
Fixed lint and build errors on Linux
Sources now build on Linux Mint 20.2 Cinnamon with respect to UE4.27
Partially addresses #63

### CODE REVIEW

File: Source/SpaceMouse/Public/SpaceMouseConfig.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 class FInputBindingManager;
 class FUICommandInfo;
+enum class EV3DCmd : uint8;
```
Comment (microdee) [inline Source/SpaceMouse/Public/SpaceMouseConfig.h:18]: forward declaring enums are not the most reliable thing to do. just do
```
#include "SmButtons.h"
```
before the generated header

**Review by microdee — CHANGES_REQUESTED** [review]
I've accepted the HIDUE PR, please update the submodule here accordingly as well.

**Comment by microdee** [comment]
Thanks for the PR's, really great stuff! Is it also working as it should with the devices on Linux? e.g.: do they control the camera in UE4 editor?

**Comment by TekuConcept** [comment]
Working on it... there seem to be some new errors to resolve relating to HIDUE. The master branch is obviously different from the referenced branch in this repo.

**Comment by TekuConcept** [comment]
> Thanks for the PR's, really great stuff! Is it also working as it should with the devices on Linux? e.g.: do they control the camera in UE4 editor?

I do not own a SpaceMouse myself. I was simply helping @Hunanbean via discord. :wink: 
The patches I submitted for this repo as well as HIDUE allow the plugin to be compiled for release-1.2.4.

File: Source/SpaceMouse/Public/SpaceMouseConfig.h (hunk 1 above)
Comment (Hunanbean) [reply Source/SpaceMouse/Public/SpaceMouseConfig.h]: I do notthink there is a SmButtons.h in this release. Perhaps Buttons.h

**Comment by Hunanbean** [comment]
It seems that neither the 2014 official 3dconnexion Linux drivers, or the Spacenavd project are sending the proper communications to the plugin. At this point, the spacemouse does not register with the plugin, but the plugin itself looks great!

**Comment by microdee** [comment]
alright then just point the HIDUE submodule to the latest change (which is your PR) and I'll merge this one too.

**Comment by microdee** [comment]
> It seems that neither the 2014 official 3dconnexion Linux drivers, or the Spacenavd project are sending the proper communications to the plugin. At this point, the spacemouse does not register with the plugin, but the plugin itself looks great!

dangit, that's a bummer :(

**Comment by Hunanbean** [comment]
Latest test, with the addition of the fix from the Prusa project,. as discovered by MalikuMane. Works quite well! 
It seems to be using the Generic HID messages. 
The spacenavd project in that scenario is only there to make Blender and other software work. Even when not installed, UE4-SpaceMouse works, When the identifier from Prusa is added. Also, when spacenavd is installed for the function of other software, it still works! So no conflicts. I hope this is clear. Please let me know if something needs clarification. Short of it is, Yay! It works! :) Thank you, David, TekuConcept and MalikuMane!

File: Source/SpaceMouse/Public/SpaceMouseConfig.h (hunk 1 above)
Comment (TekuConcept) [reply Source/SpaceMouse/Public/SpaceMouseConfig.h]: The sources compile with this header on my system

**Comment by Hunanbean** [comment]
Can someone please clarify where these changes are posted on the Git? I cannot seem to easily get TekuConcepts updated code changes, as a whole.

**Comment by microdee** [comment]
> Can someone please clarify where these changes are posted on the Git? I cannot seem to easily get TekuConcepts updated code changes, as a whole.

Here's TekuConcept's fork: https://github.com/TekuConcept/UE4-SpaceMouse

**Comment by TekuConcept** [comment]
> Can someone please clarify where these changes are posted on the Git? I cannot seem to easily get TekuConcepts updated code changes, as a whole.

`git clone https://github.com/microdee/UE4-SpaceMouse.git` - the changes are on the master branch(es).

**Comment by Hunanbean** [comment]
Thank you @TekuConcept 
I hope it is just some mistake i am making, but i am getting the following when i try to compile:
`  ERROR: Could not find definition for module 'HIDUE', (referenced via command line -> SpaceMouse.uplugin)
`

**Comment by TekuConcept** [comment]
> ` ERROR: Could not find definition for module 'HIDUE', (referenced via command line -> SpaceMouse.uplugin)`

Remember to
```
git submodule init
git submodule update
```

**Comment by Hunanbean** [comment]
I just downloaded as Zip, which is my preferred method, but i will attempt
it client way today.

Thank you 🙂

**Comment by microdee** [comment]
you have to understand these repositories do not work with downloaded zips because they have submodules in them which are not included in the zip itself. please try to familiarize yourself with basic git. it'll change your life ;)

**Comment by Hunanbean** [comment]
I got thrown off by the submodules being included in the 'release' zip i initially started this adventure with. Thanks for the directions @TekuConcept 
@microdee " it'll change your life ;)" Says the man with the frowning face emoji! ;).
Thank you both

**Comment by Hunanbean** [comment]
As expected, the build OS whitelist/blacklist is not set in the .uplugin, so that will need to be set by the end user, or errors are likely.

Regarding the Runtime issues, as you figured(microdee) there is a problem. Setting all the Runtime to Editor in the .uproject will get around that issue, allowing the project to be built and run.

**Pull request merged**